app/main.py

```diff
@@ -1,3 +1,15 @@
 def cache(func):
-    # Write your code here
-    pass
+    data = {}
+
+    def inner(*args, power: int = 0):
+        if (args, power) in data:
+            print("Getting from cache")
+            return data[args, power]
+        else:
+            print("Calculating new result")
+            if power == 0:
+                data[args, power] = func(*args)
+            else:
+                data[args, power] = func(*args, power)
+            return data[args, power]
```

Comment:
you have the same return in line 7 . change your code so that it will have return only once